util/logtools/test_timeline.ts

```diff
@@ -258,15 +258,35 @@ const testTimelineFileStartEnd = async (
     console.error(`Invalid start datetime '${args.end}', aborting.`);
     process.exit(-2);
   }
-  const allLines = fs.readFileSync(logFilePath).toString().split('\n');
+
+  const startTimestamp = startDate.getTime();
+  const endTimestamp = endDate.getTime();
+
+  const logFileStream = fs.createReadStream(logFilePath);
+
+  const logFileReadLine = readline.createInterface({
+    input: logFileStream,
+    crlfDelay: Infinity,
+  });
 
   const repo = new LogRepository();
-  const lineEvents = allLines.map((line) => ParseLine.parse(repo, line)).filter((l) => {
-    const timestamp = l?.timestamp;
+
+  const lineEvents: LineEvent[] = [];
+
+  for await (const line of logFileReadLine) {
+    const parsedLine = ParseLine.parse(repo, line);
+
+    if (parsedLine === undefined)
+      continue;
+
+    const timestamp = parsedLine?.timestamp;
+
     if (timestamp === undefined)
-      return false;
-    return timestamp >= startDate.getTime() && timestamp <= endDate.getTime();
-  }) as LineEvent[];
+      continue;
+
+    if (timestamp >= startTimestamp && timestamp <= endTimestamp)
```

Comment:
One sort of random thought on this (and it's obviously a currently-existing issues and not a result of this PR):  GameLog lines are always written with `.0000000` milliseconds.  This creates a weird situation where, ordered by timestamp, the GameLog line can predate the timestamp on the network log line that triggers it (e.g. an 0x29 line), even though it seems the GameLog line is always (?) written to the log file after the network log line (assuming there is one).  I could see this creating a weird edge case where GameLog lines that are sent in the same second we're using for `startDate` would get excluded from `lineEvents`.  The only time I could really see this being a problem is if we were writing timeline/triggers against GameLog lines as a way of starting a timeline, but I think we've basically moved away from that?

Somewhat related:  I think we use different logic in the log splitter util for extracting encounters (there, I think we do a full-text match for the start line and end line and capture everything in-between) - so it doesn't have the problem above, but it's also a lot of extra (arguably unnecessary) processing.  Maybe it's fine that we're using a different approach there? Or maybe we should align that approach to what's being done here with a startDate/endDate timestamp?  Not sure I really have an opinion, more just wanted to throw that out there.

Reply:
Yeah, we'll sometimes end up with missing `GameLog` lines at the start and extra `GameLog` lines at the end.

The reason that `GameLog` lines are always truncated to a full second is because the struct that Ravahn is reading for them in memory doesn't store milliseconds.

A more reasonable approach when using a datetime like this would be to skip to the first line with that datetime matching, then include all lines until a line that's after the end datetime is encountered.

The whole reason I fixed this logic to begin with is because the `-lf` flags for fight detection didn't properly find the fights in this logfile, which is an entirely separate issue.

If anything, longer-term, I'd rather flags for start line and end line and use those for these sort of edge cases, since that's a bit more exact than the timestamp-based logic.

Reply:
> If anything, longer-term, I'd rather flags for start line and end line and use those for these sort of edge cases, since that's a bit more exact than the timestamp-based logic.

Makes sense - maybe create a follow-up issue if you're so inclined?  Otherwise, LGTM.